OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -321,12 +323,39 @@ protected virtual void Tick(Actor self)
 
 			var oldCachedPosition = cachedPosition;
 			cachedPosition = self.CenterPosition;
-			isMoving = (oldCachedPosition - cachedPosition).HorizontalLengthSquared != 0;
-			isMovingVertically = (oldCachedPosition - cachedPosition).VerticalLengthSquared != 0;
+			UpdateIsMoving(self, (oldCachedPosition - cachedPosition).HorizontalLengthSquared != 0, (oldCachedPosition - cachedPosition).VerticalLengthSquared != 0);
 
 			Repulse();
 		}
 
+		void UpdateIsMoving(Actor self, bool isMovingNow, bool isMovingVerticallyNow)
+		{
+			// NOTE: Traits implementing INotifyMoving may need to evaluate IsMoving or IsMovingVertically from the other's notification methods.
+			// This is problematic, as they cannot be updated at the exact same moment. If, for example, INotifyMoving.StartedMoving
+			// were triggered first and evaluated IsMovingVertically before the latter has been updated, it would potentially make
+			// the wrong decision if both IsMoving and IsMovingVertically changed their value on the same tick.
+			// Therefore, updating both IsMoving *and* IsMovingVertically *before* any of the trait notifications are triggered is a must-have,
+			// at least for movement traits like Aircraft were both booleans play a role.
```

Comment:
Considering this, IMO it really doesn't make sense to have separate notifications for horizontal vs vertical movement. Can we instead define a `[Flags] public enum MovementType { None, Horizontal, Vertical }` and then have one set of start/stop callbacks for both?